### PR TITLE
S3, Http config; looping polling for SNMP

### PR DESCRIPTION
**Pull request by i3149**
Allow snmp config files to be prefixed s3:// or http(s):// in addition to a local path. If a remote path, fetch the resource from the remote place. 

Also, adds a flag `--snmp_discovery_min` which will run discovery every X minutes and restart the snmp polling system if a new device is found.

### CODE REVIEW

File: pkg/inputs/snmp/disco.go [hunk 1]
```diff
@@ -248,7 +276,7 @@ func doubleCheckHost(result scan.Result, timeout time.Duration, ctl chan bool, m
 	foundDevices[result.Host.String()] = &device
 }
 
-func addDevices(foundDevices map[string]*kt.SnmpDeviceConfig, snmpFile string, conf *kt.SnmpConfig, isTest bool, log logger.ContextL) error {
+func addDevices(ctx context.Context, foundDevices map[string]*kt.SnmpDeviceConfig, snmpFile string, conf *kt.SnmpConfig, isTest bool, log logger.ContextL) (error, int, int, int) {
```
Comment (ehazlett) [inline pkg/inputs/snmp/disco.go:279]: Should we do something like create a struct for the return?  Something like
```type snmpDevices {
      added int
      replaced int
      delta int
}
```
and then return `(*snmpDevices, error)`?  It makes it a bit more readable IMO.  I know I'll forget this order down the road ;)

File: pkg/inputs/snmp/snmp.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ var (
 	snmpWalkFormat = flag.String("snmp_walk_format", "", "use this format for walked values if -snmp_do_walk is set.")
 	snmpOutFile    = flag.String("snmp_out_file", "", "If set, write updated snmp file here.")
 	snmpPollNow    = flag.String("snmp_poll_now", "", "If set, run one snmp poll for the specified device and then exit.")
+	snmpDiscoDur   = flag.Int("snmp_discovery_sec", 0, "If set, run snmp discovery on this interval (in seconds).")
```
Comment (thezackm) [inline pkg/inputs/snmp/snmp.go:36]: recc updating this to `snmp_discovery_min` to allow a more user friendly scheduling experience

Comment (i3149) [reply pkg/inputs/snmp/snmp.go]: good call.

File: pkg/inputs/snmp/disco.go (hunk 1 above)
Comment (i3149) [reply pkg/inputs/snmp/disco.go:279]: hah, totally.

**Comment by i3149** [comment]
OK -- thanks for the pre-comments. I think this is ready to look at generally.

**Review by ehazlett — APPROVED** [review]

**Comment by i3149** [comment]
@araddas -- any opinions on this one?

**Comment by araddas** [comment]
@i3149 im on PTO, will get back to you next week.

**Comment by araddas** [comment]
@i3149 this looks good, shoot off a build and I will test it out.

**Review by jbeveland27 — APPROVED** [review]

**Comment by i3149** [comment]
> @i3149 this looks good, shoot off a build and I will test it out.

Cool, give it a shot now when you get a chance.

**Pull request merged**